src/redux/actions/progressions/synchronize.js

```diff
@@ -28,65 +30,62 @@ export type Action =
       meta: {|id: string|}
     |}>;
 
-export const synchronizeProgression = (progressionId: string): StoreAction<Action> => {
-  return async (dispatch, getState, options) => {
-    await dispatch({
-      type: SYNCHRONIZE_REQUEST,
-      meta: {id: progressionId}
-    });
+export const synchronizeProgressions: StoreAction<Action> = async (dispatch, getState, options) => {
+  const {services} = options;
+  const state = getState();
+  const token = getToken(state);
+  const brand = getBrand(state);
 
-    const state = getState();
-    const token = getToken(state);
-    const brand = getBrand(state);
+  if (token === null) throw new TypeError('Token not defined');
+  if (brand === null) throw new TypeError('Brand not defined');
 
-    const {services} = options;
+  const progressions = await services.Progressions.getAll();
+  const synchronizedProgressionsIds: Array<string> = await services.Progressions.getSynchronizedProgressionIds();
 
+  const syncProgression = async (progression: Progression, countNumberOfRetries?: number = 0) => {
```

Comment:
```suggestion
  const syncProgression = async (progression: Progression, countNumberOfRetries?: number = 5) => {
```

Si on inverse votre logique, la configuration du nombre de retry est en haut de la fonction. C'est surement mieux d'avoir le maximum d'info dans l'entete de la function.